Use named volumes for MySQL volume

beeradb opened the pull request:
## The Problem:
This is an alternative approach to #337 which uses named volumes instead of host mounted directories.

It allows MySQL data to persist even when a site is removed using `ddev rm`, but rather than host mounting the files, it uses the native docker volume functionality, https://docs.docker.com/compose/compose-file/#volume-configuration-reference. Personally, I think it will be easier to maintain in the long run if we just rely on what docker is giving us. That being said, I can also understand the potential UX benefits of exposing the volume as a directory.

I mostly put this together to prove to myself it would work as I expected and from there the changes were small enough that it made sense to just open a PR to compare the approaches.

## The Test:
- Start a site
- Install drupal/wordpress
- Remove the site
- Start the site again
- Observe previously installed Drupal/WP site still active.

you should also be able to change MySQL container versions without any data loss.

If we stick with this method, it might make sense to expand the `ddev rm` command to have an option for cleaning up the volume as well, in case users really want to start completely from scratch

## Automation Overview:
I removed the previous tests around the `-y` flag, as ddev rm is no longer a destructive operation.

## Related Issue Link(s):
#208 #337

beeradb commented:
Doing some performance testing here on a vanilla d8 installation. I did two passes:

- Pass 1: `0.66s user 0.15s system 0% cpu 1:47.35 total`
- Pass 2: `0.63s user 0.14s system 0% cpu 1:46.18 total`

tannerjfco commented:
This approach seems to work well from my testing, and we don't incur any performance penalties going this route, which is preferable even if the penalty was determined to be negligible. I don't think I have any objections taking this approach, but we do need to figure out how data will be destroyed if desired. With the host mounted approach, we could potentially just instruct users to delete the directory to destroy. With volumes, we likely need a flag of some sort on the remove command, though I would have found one useful for the host-mounted approach as well.

beeradb commented:
Based on initial feedback from @tannerjfco and @rickmanelius it feels like there is some momentum for this direction. As such, I've updated the PR to include the following items:

- A new flag for ddev rm, `--remove-data` which includes the -v command when running docker-compose down to clean up old values. **This doesn't currently remove settings.php, but perhaps it should?**
- Updated help text for ddev RM to reflect the new behavior.
- `removeSites()` which occurs at the end of test teardown now includes the `--remove-data` flag. It also has a check to ensure the volumes are properly removed
- a nginx-logs volume. It already was a volume, all this does is gives is a nice name for anyone poking around in `docker volume list`. We could remove this as a volume as well, the only real benefit of having it as one is that logs get persisted between container restarts, which isn't a particularly high-value feature.

rickmanelius commented:
Other than my one requested change, I think the updated flag and helper text looks good.

tannerjfco commented:
I have added the requested functionality to this PR to remove ddev-generated settings files on remove.

rickmanelius commented:
Noting that I'm beginning a test.

rickmanelius commented:
## Binary Setup

* Testing https://github.com/drud/ddev/pull/340
* Assets https://circleci.com/gh/drud/ddev/1152
* Binary ddev_osx.v0.7-19-gc88cf176.zip
* Run `ddev version`

➜  ~ ddev version
cli:       	v0.7-19-gc88cf176
web:       	drud/nginx-php-fpm7-local:v0.7.3
db:        	drud/mysql-local-57:v0.5.0
dba:       	drud/phpmyadmin:v0.2.0
router:    	drud/ddev-router:v0.4.3
commit:    	v0.7-19-gc88cf176
build info:	Built Tue Jun 20 17:22:22 UTC 2017 circleci@default-1497979176912 drud/golang-build-container:v0.4.2

## App Setup

* Clean git@github.com:drud/wordpress.git
* Removed the .ddev folder
* Re-ran ddev configuration
* ddev start

## Scenario Tests

Here are the 4 things to test.

1. Allow ddev to manage config, remove site and confirm db is there.
2. Allow ddev to manage config, remove site w/flag and confirm db and settings are gone.
3. Override ddev to manage config, remove site and confirm db is there.
4. Override ddev to manage config, remove site w/flag and confirm db is gone while settings remains.

See https://github.com/drud/ddev/issues/308 for details on removing settings.php and wp-config.php

### Test 1

* Finished the WordPress site install
* `ddev rm`
* ISSUE: wp-config.php is still there!
* `ddev start`
* Confirmed that the database is present!

### Test 2

* Using the test run from Test 1.
* `ddev rm --remove-data`
* Confirmed I see 
  * Removing volume ddevwordpress_nginx-logs
  * Removing volume ddevwordpress_mysql
* ISSUE: wp-config.php is still there!
* `ddev start`
* Confirmed that I'm at a default install page.

## NOTES

* Please note that I'm not seeing ddev-generated in wp-config.php or settings.php as per https://github.com/drud/ddev/blob/5f0730f429088847d5204b0408129f3ed7cb5dbf/pkg/cms/config/drupal.go#L18. Therefore, I can't proceed on the other tests.

rickmanelius commented:
@beeradb Given that Tanner reviewed your previous work and that I already reviewed some of Tanner's additions, any objection to merging?
